### PR TITLE
fix(dev-workflow-v2): harden review and PR feedback workflows

### LINKED ISSUE #260: Introduce test-reviewer agent into code review process
## References
PR #259 reflection — CodeRabbit caught missing rename test that local reviewers missed

## Summary
Split test review responsibilities from the main code-review agent into a dedicated test-reviewer agent. The current code-review agent handles both design/convention enforcement and test quality assessment, making it less focused on each. A dedicated test-reviewer would specialize in test completeness, edge case coverage, and format-specific test requirements like RFC-017.

## Full Details
Create a new .claude/agents/test-reviewer.md agent that focuses exclusively on test quality. Include rules for: (1) external format parser test completeness (all documented variants), (2) edge case coverage per RFC-014, (3) success-path assertions per RFC-013, (4) mock cleanup per RFC-016. Wire it into the code review step in run-code-review.ts alongside existing reviewers.

## Acceptance Criteria
- [ ] New .claude/agents/test-reviewer.md agent created with focused test quality rules
- [ ] Agent includes RFC-017 structured format parsing test requirement
- [ ] Agent wired into run-code-review.ts reviewer list
- [ ] Existing code-review agent updated to remove overlapping test responsibilities
- [ ] All tests pass with new reviewer in pipeline

### PR DESCRIPTION
**Pull request by NTCoding**
## Description
This PR hardens the `dev-workflow-v2` review lifecycle across both OpenCode and GitHub review handling. It keeps the review flow deterministic, makes pull request descriptions explicitly link the issue they close, and replaces the old manual PR feedback state with an automatic wait-for-CodeRabbit flow followed by live mergeability verification.

## Linked Issue
Closes #260

## What Problem Does This PR Solve?
The existing branch work left the review workflow fragmented in two places: OpenCode review agents were not fully integrated into the structured review state, and PR feedback handling could advance without actually waiting for CodeRabbit or proving that the PR had become mergeable again. The PR body format also did not require a linked issue, which weakened GitHub and CodeRabbit issue tracking.

## Key Changes
- Enable the OpenCode review subagents and tighten REVIEWING so verdicts cannot be recorded before a reviewer has actually returned a result
- Require a richer PR body template with a linked issue section and explicitly enable `reviews.assess_linked_issues` in CodeRabbit config
- Replace `CHECKING_FEEDBACK` with automatic `AWAITING_PR_FEEDBACK` polling that waits for CodeRabbit, then routes to `ADDRESSING_FEEDBACK`, `REFLECTING`, or `BLOCKED`
- Rework `ADDRESSING_FEEDBACK` so the agent fetches live GitHub review state, and only leaves the state after a workflow verification confirms there are no unresolved actionable threads and no `CHANGES_REQUESTED` review decision
- Add full workflow, CLI, and GitHub client coverage for the new behavior

## Notable Architectural Changes / Impact
The workflow state machine now owns PR feedback waiting as a synchronous automatic side effect in `workflow.ts`, while `ADDRESSING_FEEDBACK` verifies mergeability against live GitHub state instead of relying on manual unresolved-thread counts. This keeps the workflow deterministic without mirroring full PR thread state locally.

## Validation
- `npm run typecheck` in `tools/dev-workflow-v2`
- `npm test` in `tools/dev-workflow-v2`

## Notes
- The current branch PR already contained the earlier OpenCode review-agent work, so this PR body describes the full branch scope after the new PR feedback fixes were added.

## Summary by CodeRabbit

* **New Features**
  * Automated PR feedback polling with timeout and new verify-feedback-addressed operation
  * Added /dev-workflow-v2:workflow show-state for inspecting workflow state
  * Dynamic registration of review subagents

* **Documentation**
  * Renamed "Checking Feedback" → "Awaiting PR Feedback" and updated state diagrams, transitions, and guidance
  * Enhanced PR creation and feedback-addressing instructions

* **Chores**
  * Enabled CodeRabbit linked-issue assessment
  * Bumped plugin version to 0.0.26
  * Updated ShellCheck usage in CI and improved test fixtures/configuration

### CODE REVIEW

**Comment by NTCoding** [comment]
@coderabbitai review

File: tools/dev-workflow-v2/src/features/workflow/entrypoint/workflow-cli-transitions.spec.ts [hunk 1]
```diff
@@ -24,8 +24,12 @@ describe('workflow-cli transitions', () => {
       readonly hasCommitsVsDefault: boolean
       readonly workingTreeClean: boolean
     }>
+    readonly getPrFeedback?: TestContext['workflowDeps']['getPrFeedback']
   }): TestContext {
-    const ctx = buildTestContext()
+    const ctx =
+      overrides?.getPrFeedback === undefined
+        ? buildTestContext()
+        : buildTestContext({ getPrFeedback: overrides.getPrFeedback })
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply tools/dev-workflow-v2/src/features/workflow/entrypoint/workflow-cli-transitions.spec.ts]: ✅ Fixed in later commits. The setup helper no longer uses the old pass-through shape from this stale thread.

File: tools/dev-workflow-v2/states/addressing_feedback.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- [ ] Respond to each thread using gh CLI:
+  ```bash
+  # Reply to thread (use ✅ **Fixed** or ❌ **Rejected** prefix)
+  gh api graphql -f query='mutation { addPullRequestReviewThreadReply(input: {pullRequestReviewThreadId: "<THREAD_ID>", body: "<PREFIX>: <explanation>"}) { comment { id } } }'
+  # Resolve thread
+  gh api graphql -f query='mutation { resolveReviewThread(input: {threadId: "<THREAD_ID>"}) { thread { id } } }'
+  ```
+- [ ] Commit all fixes
+- [ ] Re-fetch the PR feedback from GitHub and confirm there are no unresolved actionable threads and no `CHANGES_REQUESTED` review decision
+- [ ] Verify the PR is mergeable from a review perspective: `/dev-workflow-v2:workflow record-feedback-addressed`
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply tools/dev-workflow-v2/states/addressing_feedback.md]: ✅ Fixed in later commits. The feedback step wording now states that the command both records and verifies live GitHub state.

File: tools/dev-workflow-v2/README.md [hunk 3]
```diff
@@ -38,6 +38,14 @@ Renames the worktree branch to match the issue, reads the issue details, initial
 
 Low-level state machine CLI. Used by the other commands and state instructions — not called directly by users.
 
+Useful internal command for state-driven instructions:
+
+```bash
+/dev-workflow-v2:workflow show-state
+```
+
+This returns the current workflow state as JSON so state instructions can extract exact values such as `githubIssue`, `prNumber`, and `taskCheckPassed` without guessing.
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply tools/dev-workflow-v2/README.md:47]: ✅ Fixed. The docs now use the platform built-in `get-state` command instead of the nonexistent `show-state` alias.

File: tools/dev-workflow-v2/src/features/workflow/domain/states/reflecting.ts [hunk 4]
```diff
@@ -1,18 +1,9 @@
...
-  pass, fail 
-} from '@nt-ai-lab/deterministic-agent-workflow-dsl'
 
 export const reflectingState: ConcreteStateDefinition = {
   emoji: '🪞',
   agentInstructions: 'states/reflecting.md',
   canTransitionTo: ['COMPLETE', 'BLOCKED'],
-  allowedWorkflowOperations: ['record-reflection'],
+  allowedWorkflowOperations: [],
   forbidden: { write: true },
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply tools/dev-workflow-v2/src/features/workflow/domain/states/reflecting.ts:8]: ❌ Rejected. The workflow no longer owns reflection persistence. Reflection is recorded exclusively through the platform reflection store, so no workflow-level reflection guard is being added.

File: tools/dev-workflow-v2/src/features/workflow/domain/workflow-addressing-feedback.spec.ts [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    expect(outsideState.result.pass).toStrictEqual(false)
+    expect(noPrNumber.result).toMatchObject({
+      pass: false,
+      reason: expect.stringContaining('prNumber not set'),
+    })
+    expect(fetchError.result).toMatchObject({
+      pass: false,
+      reason: expect.stringContaining('Unable to fetch PR feedback'),
+    })
+  })
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply tools/dev-workflow-v2/src/features/workflow/domain/workflow-addressing-feedback.spec.ts:146]: ✅ Fixed. The outside-state assertion now checks both `pass: false` and the operation-gate failure reason substring.

File: tools/dev-workflow-v2/states/addressing_feedback.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# ADDRESSING_FEEDBACK State
+
+You are addressing PR review feedback.
+
+Start by running `/dev-workflow-v2:workflow show-state` and extracting `prNumber` from its JSON output, then fetch the current PR feedback directly from GitHub for that PR.
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply tools/dev-workflow-v2/states/addressing_feedback.md]: ✅ Fixed in later commits. The instructions now use the platform built-in `get-state` command instead of `show-state`.
